src/core/control/Control.cpp

```diff
@@ -2979,6 +2992,27 @@ void Control::clipboardPasteXournal(ObjectInputStream& in) {
     }
 }
 
+void Control::moveSelectionToLayer(size_t layerNo) {
+    PageRef currentP = getCurrentPage();
+    if (layerNo >= currentP->getLayerCount()) {
+        return;
+    }
+    auto selection = getWindow()->getXournal()->getSelection();
+    if (!selection) {
+        return;
+    }
+
+    auto* oldLayer = currentP->getSelectedLayer();
+    auto* newLayer = currentP->getLayers()->at(layerNo);
+    auto moveSelUndo = std::make_unique<MoveSelectionToLayerUndoAction>(currentP, getLayerController(), oldLayer, currentP->getSelectedLayerId() - 1, layerNo);
+    for (auto* e : selection->getElements()) {
+        moveSelUndo->addElement(newLayer, e, newLayer->indexOf(e));
+    }
+    undoRedo->addUndoAction(std::move(moveSelUndo));
+
+    getLayerController()->switchToLay(layerNo + 1, false, false);
```

Comment:
nit: the optional parameters should be commented:
```suggestion
    getLayerController()->switchToLay(layerNo + 1, /*hideShow=*/false, /*clearSelection=*/false);
```